File: BOJ/6550.py
import sys

while True:
    line = sys.stdin.readline().strip()
    if not line:
        break
    s, t = line.split()
    idx = 0
    chk = 0

    for i in range(len(t)):
        if t[i] == s[idx]:
            idx += 1
            if idx == len(s):
                chk = 1
                break

    if chk == 0:
        print("No")
    else:
        print("Yes")

File: BOJ/6550.js
let input = require("fs")
  .readFileSync("예제.txt")
  .toString()
  .trim()
  .split("\n");

input.forEach((el) => {
  let a = el.split(" ");
  let idx = 0;
  let chk = 0;

  for (let i = 0; i < a[1].length; i++) {
    if (a[1][i] === a[0][idx]) {
      idx++;
      if (idx === a[0].length) {
        chk = 1;
        break;
      }
    }
  }

  if (chk === 0) console.log("No");
  else console.log("Yes");
});
